### ISSUE TITLE
Receiving messageID (.getId()) when using .async()?

### ISSUE THREAD

**Issue by JackHewson** [bbottema/simple-java-mail#338]
According https://www.simplejavamail.org/features.html#section-sent-id it is possible to receive the generated messageId by:

``` 
mailer.sendMail(email); // id updated during sending!
email.getId(); // <1420232606.6.1509560747190@Cypher>`
``` 

**But how is it possible to receive the messageId when using `.async()`?**
When using `.async()`, the `email.getId();` is  `null`.

Thank you!

**Comment by bbottema** [comment]
As the code to update and send is run asynchronously, you would have to use the completion handlers to know when the email was actually updated and set. These is currently only available from the method `mailer.send(email, true)`:

```java
mailerSMTPBuilder.buildMailer().sendMail(emailNormal, true)
	.onSuccess(() -> {});
```
or
```java
mailerSMTPBuilder.buildMailer().sendMail(emailNormal, true).getFuture().get()
```
However, it does make sense to return the same from `mailer.send(email)`, since it can now be configured on the builders. I'll think about adding that.

**Comment by bbottema** [comment]
I've just released 6.7.5 which now returns AsyncResponse from mailer.sendMail(singleArgument). That should make it easier for you to handle the async exeution.

**Comment by bbottema** [comment]
Current 9.x usage (verified against 9.3.2):

```java
Mailer mailer = MailerBuilder
    .withSMTPServer("smtp.example.com", 587)
    .async()
    .buildMailer();

CompletableFuture<Void> result = mailer.sendMail(email);
result.whenComplete((unused, failure) -> {
    if (failure != null) {
        System.err.println(failure.getMessage());
    }
});
```

The one-argument call still honors the mailer's configured async default, but `AsyncResponse` was replaced by `CompletableFuture` in 7.0. Since 9.2, the future carries failures from the complete async operation. See the [7.0 migration](https://www.simplejavamail.org/migration-notes-7.0.0.html#completable-futures), [9.2 failure-reporting change](https://www.simplejavamail.org/migration-notes-9.2.0.html#async-failure-reporting), and [current async-result documentation](https://www.simplejavamail.org/features.html#section-handling-async-result).
